cmake/Codegen.cmake

```diff
@@ -53,15 +53,23 @@ install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
 
 # ---[ ATen specific
 if (INTERN_BUILD_ATEN_OPS)
-  SET(OPT_FLAG "-O3 ")
   IF(MSVC)
-    SET(OPT_FLAG "/Ox /fp:strict ")
-  ENDIF()
-  SET(VCOMP_LIB "vcomp")
-
-  IF("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
-    SET(OPT_FLAG " ")
-    SET(VCOMP_LIB "vcompd")
+	SET(OPT_FLAG 
+		$<$<CONFIG:Debug>:"${CMAKE_CXX_FLAGS_DEBUG}">
```

Comment:
Why do you introduce `CMAKE_CXX_FLAGS_XXX` here? The OPT_FLAG should be set to `/Ox /fp:strict` for non-debug builds and ` ` for debug builds, right?

Reply:
The defaults when using MSVC come with a whole set of specific optimizations flags per configuration that are not actually `/Ox` by default in any case : 
```
 -- CMAKE_CXX_FLAGS= /DWIN32 /D_WINDOWS /W3 /GR /EHsc
 -- CMAKE_CXX_FLAGS_DEBUG= /MDd /Zi /Ob0 /Od /RTC1
 -- CMAKE_CXX_FLAGS_MINSIZEREL= /MD /O1 /Ob1 /DNDEBUG
 -- CMAKE_CXX_FLAGS_RELEASE= /MD /O2 /Ob2 /DNDEBUG
 -- CMAKE_CXX_FLAGS_RELWITHDEBINFO= /MD /Zi /O2 /Ob1 /DNDEBUG
```
note that `/Ox` is almost the same as `/O2` but with additional things included (https://docs.microsoft.com/en-us/cpp/build/reference/o-options-optimize-code?view=vs-2019)

I feel it is better to stick with the cmake default (or anything overridden by the user), with the exception of `/fp:strict` that we enforce.

Reply:
I believe those flags should be added by default, so looks like you can simply remove those stuff and keep `/fp:strict` only.